Assets/_Laz/Scripts/Managers/CheckPointManager.cs

```diff
@@ -1,28 +1,85 @@
-using System.Collections.Generic;
+using PerigonGames;
 using UnityEngine;
 
 namespace Laz
 {
     public class CheckPointManager : MonoBehaviour
     {
-        [SerializeField] private List<Transform> _checkPoints = null;
-        public Transform LastCheckPoint { get; private set; } = null;
+        [SerializeField] private Checkpoint[] _checkpoints;
+        private Checkpoint _activeCheckpoint = null;
+        private Vector3 _initialCheckPointPosition = Vector3.zero;
 
         public void Initialize()
         {
-            UpdateCheckPoint();
+            if (_checkpoints.IsNullOrEmpty())
```

Comment:
Instead of returning, you should put this check in 
`OnAwake` and then throw an error. 

Since i'm assuming you must have at least 1 checkpoint for the game to work?

Reply:
You should put this OnAwake()

Fail Faster better than Fail maybe later. 

If Will comes in and puts this CheckpointManager in a new scene, it won't fail until (on the off chance) he calls the public methods. 

So better to fail before the game even starts so he can fix everything off the bat